tests/kafkatest/services/connect.py

```
@@ -252,6 +256,14 @@ def _rest_with_retry(self, path, body=None, node=None, method="GET", retries=40,
     def _base_url(self, node):
         return 'http://' + node.account.externally_routable_ip + ':' + str(self.CONNECT_REST_PORT)
 
+    def append_to_environment_variable(self, envvar, value):
```

what is the logic for this, including the different ways `env_opts` is being encoded?

---

I ended up adding the heap jvm arguments in `KAFKA_OPTS` instead of `KAFKA_HEAP_OPTS` because the latter has defaults for sizes if it's empty. However, if I needed to add values to an env variable, I can't just append them to it's current value, because the value is wrapped with `\"` 

That's why I'm adding this method - unused here - that can be used by tests that inherit this class. Let me know if you think I should remove.

---

yeah, in general the construction of commands could be cleaned up quite a bit for extensibility -- i think we tend to forget about the importance/value of doing this when writing tests in the Kafka repo, but these classes should be reusable and extensible. this approach seems pretty error-prone, although at least it should consistently fail if the quoting gets messed up since you'll have invalid extra args on the command line.